model/project.rb

```diff
@@ -41,6 +41,24 @@ def path
     "/project/#{ubid}"
   end
 
+  def has_resources
+    access_tags_dataset.exclude(hyper_tag_table: [Account.table_name.to_s, Project.table_name.to_s, AccessTag.table_name.to_s]).count > 0
+  end
+
+  # Still thinking about the name, open for suggestions
+  def destroy_tags_and_hide
+    DB.transaction do
+      access_tags.each { |access_tag| access_tag.applied_tags_dataset.destroy }
```

Comment:
nit: No need to destroy `applied_tags` manually. They are destroyed automatically already

https://github.com/ubicloud/ubicloud/blob/4c29c74cc5c367a3dd0fab81de71c6e69d6a92f7/model/access_tag.rb#L9